src/control/cmd/daos/pool.go

```diff
@@ -251,17 +280,33 @@ func (cmd *poolQueryCmd) Execute(_ []string) error {
 	pinfo := C.daos_pool_info_t{
 		pi_bits: dpiQueryAll,
 	}
-	rc := C.daos_pool_query(cmd.cPoolHandle, nil, &pinfo, nil, nil)
+	if cmd.ShowDisabledRanks {
+		pinfo.pi_bits &= C.uint64_t(^(uint64(C.DPI_ENGINES_ENABLED)))
+	}
+
+	rc := C.daos_pool_query(cmd.cPoolHandle, ranklistPtr, &pinfo, nil, nil)
 	if err := daosError(rc); err != nil {
+		C.d_rank_list_free(ranklist)
```

Comment:
does rank list get allocated by daos_pool_query() if it returns rc != 0?
I think  d_rank_list_free() checks for a NULL pointer anyway so maybe it's fine to just always call it before returning from this function.

another idea I had is whether the Go defer statement could be used to ensure that d_rank_list_free() is called before returning, without having to insert it in all of the error conditional branches